File: cs50x/2week/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float letter, word, sentence;

int readability(string text);

int main(void)
{
    //taking input
    string text = get_string("Text: ");

    //using function
    int grade = readability(text);

    //print result
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

    //tests
    //printf("%0.0f--%0.0f--%0.0f\n", letter, word, sentence);
    //printf("%f--%f\n", L, S);
}


int readability(string text)
{
    letter = 0.0, word = 1.0, sentence = 0.0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letter++;
        }
        else if (isspace(text[i]))
        {
            word++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentence++;
        }
    }
    float L = letter / word * 100, S = sentence / word * 100, index = 0.0588 * L - 0.296 * S - 15.8;

    return round(index);
}

File: cs50x/6week/readability/readability.py
from cs50 import get_string


def main():

    # taking input
    text = get_string("Text: ")

    # using function
    grade = readability(text)

    # print result
    if (grade >= 16):
        print("Grade 16+")
    elif (grade < 1):
        print("Before Grade 1")
    else:
        print("Grade", grade)


def readability(text):

    letter, word, sentence = 0, 1, 0

    for i in range(0, len(text)):
        if (text[i].isalpha()):
            letter += 1
        elif (text[i].isspace()):
            word += 1
        elif (text[i] == '.' or text[i] == '?' or text[i] == '!'):
            sentence += 1

    L, S = (letter / word * 100), (sentence / word * 100)
    index = (0.0588 * L - 0.296 * S - 15.8)

    return round(index)


if __name__ == "__main__":
    main()